code/modules/hivemind/mobs.dm

```diff
@@ -0,0 +1,913 @@
+//how to use: Make hive mob, then just use this one and don't forget to delete victim
+
+/mob/living/simple_animal/hostile/hivemind/resurrected/proc/absorb_the(mob/living/victim)
+	icon = victim.icon
+	icon_state = victim.icon_state
+	//simple_animal's change their icons to dead one after death, so we make special check
+	if(istype(victim, /mob/living/simple_animal))
+		var/mob/living/simple_animal/SA = victim
+		icon_state = SA.icon_living
+		icon_living = SA.icon_living
+		speed = SA.speed + 3 //why not?
+		attacktext = SA.attacktext
+
+	//another check for superior mobs, fuk this mob spliting
+	if(istype(victim, /mob/living/carbon/superior_animal))
+		var/mob/living/carbon/superior_animal/SA = victim
+		icon_state = SA.icon_living
+		icon_living = SA.icon_living
+		attacktext = SA.attacktext
+
+	//now we work with icons, take victim's one and multiply it with special icon
+	var/icon/infested = new /icon(icon, icon_state)
+	var/icon/covering_mask = new /icon('icons/mob/hivemind.dmi', "covering[rand(1, 3)]")
+	infested.Blend(covering_mask, ICON_MULTIPLY)
+	overlays += infested
+
+	maxHealth = victim.maxHealth * 2 + 10
+	health = maxHealth
+	name = "[pick("rebuilded", "undead", "unnatural", "fixed")] [victim.name]"
+	if(lentext(victim.desc))
+		desc = desc + " But something wasn't right..."
+	density = victim.density
+	mob_size = victim.mob_size
+	pass_flags = victim.pass_flags
+
+	//corrupted speak imitation
+	var/phrase_amount = rand(2, 5)
+	for(var/count = 1 to phrase_amount)
+		var/first_word = pick("You", "I", "They", "Hive", "Corpses", "We", "Your friend", "This ship", "Your mind", "These guys")
+		var/second_word = pick("kill", "stop", "transform", "connect", "rebuild", "fix", "hug", "hit", "told", "help", "rework", "burn")
+		var/third_word = pick("them", "me", "you", "your soul", "us", "hive", "system", "this ship", "your head", "your brain")
+		var/end_symbol = pick("...", ".", "?", "!")
+		var/phrase = "[first_word] [second_word] [third_word][end_symbol]"
+		speak.Add(phrase)
+
+
+/mob/living/simple_animal/hostile/hivemind/resurrected/death()
+	..()
+	gibs(loc, null, /obj/effect/gibspawner/robot)
+	qdel(src)
+
+///we live to/////////////////////////////////////////////////////////////////
+////////////////////////////////////SMALL GUYS///////////////////////////////
+//////////////////////////////////////////////////////////////die for hive//
+
+
+/////////////////////////////////////STINGER//////////////////////////////////
+//Special ability: none
+//Just another boring mob without any cool abilities
+//High chance of malfunction
+//Default speaking chance
+//Appears from dead small mobs or from hive spawner
+//////////////////////////////////////////////////////////////////////////////
+
+/mob/living/simple_animal/hostile/hivemind/stinger
+	name = "medbot"
+	desc = "Ah-h little good guy. Wait a minute, is this a blade? Yep. You see a whole blade in place where syringe must be."
```

Comment:
"A little medical robot. He looks somewhat underwhelmed. Wait a minute, is that a blade?"